### PR TITLE
[DLR] Support multithreading (experimental)

### PR DESCRIPTION
**Pull request by stu1130**
## Channel for questions ##
Before or while filing an issue please feel free to join our [Slack channel](https://join.slack.com/t/deepjavalibrary/shared_invite/zt-ar91gjkz-qbXhr1l~LFGEIEeGBibT7w) to get in touch with development team, ask questions, find out what's cooking and more!

## Description ##
(Brief description of what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- [ ] Code is well-documented: 
    - For user-facing API changes, Java doc has been updated. 
    - For new examples, README.md is added to explain the what the example does.
- [ ] To the my best knowledge, [examples](https://github.com/awslabs/djl/tree/master/examples) and [jupyter notebooks](https://github.com/awslabs/djl/tree/master/jupyter) are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, Java doc)
- [ ] Feature2, tests, (and when applicable, Java doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

File: dlr/dlr-engine/src/main/java/ai/djl/dlr/engine/DlrSymbolBlock.java [hunk 1]
```diff
@@ -66,11 +66,13 @@ public NDList forward(
         // TODO maybe verify the number of inputs
         // currently we assume the order of the input NDList is the same
         // as the model input
-        for (int i = 0; i < inputs.size(); ++i) {
-            JniUtils.setDlrInput(modelHandle, inputs.get(i), i);
+        synchronized (DlrSymbolBlock.class) {
```
Comment (lanking520) [inline dlr/dlr-engine/src/main/java/ai/djl/dlr/engine/DlrSymbolBlock.java:69]: This is not the true multi-threading. More like a thread-safe forward method.

File: examples/build.gradle [hunk 2]
```diff
@@ -67,6 +70,8 @@ task benchmark(type: JavaExec) {
                         environment("OMP_NUM_THREADS", "1")
                         environment("TF_NUM_INTEROP_THREADS", "1")
                         environment("TF_NUM_INTRAOP_THREADS", "1")
+                    } else if (System.properties["ai.djl.default_engine"] == "DLR") {
+                        environment("TVM_NUM_THREADS", Runtime.getRuntime().availableProcessors())
```
Comment (lanking520) [inline examples/build.gradle:74]: This doesn't do anything right now?

File: examples/src/main/java/ai/djl/examples/inference/benchmark/util/AbstractBenchmark.java [hunk 3]
```diff
@@ -288,10 +287,7 @@ public NDList processInput(TranslatorContext ctx, Object input) {
                         /** {@inheritDoc} */
                         @Override
                         public Object processOutput(TranslatorContext ctx, NDList list) {
-                            FloatBuffer fb = list.get(0).toByteBuffer().asFloatBuffer();
-                            float[] ret = new float[fb.remaining()];
-                            fb.get(ret);
-                            return ret;
+                            return list.get(0).toFloatArray();
```
Comment (frankfliu) [inline examples/src/main/java/ai/djl/examples/inference/benchmark/util/AbstractBenchmark.java:290]: the output is not always float32, we need the original code

File: examples/build.gradle (hunk 2 above)
Comment (frankfliu) [reply examples/build.gradle:74]: If we assume DLR models are lightweight, should we give it 2X of cores?

File: dlr/dlr-engine/src/main/java/ai/djl/dlr/engine/DlrSymbolBlock.java (hunk 1 above)
Comment (stu1130) [reply dlr/dlr-engine/src/main/java/ai/djl/dlr/engine/DlrSymbolBlock.java:69]: right as it don't support real multithreading if we don't want to copy model & dlr lib

File: examples/build.gradle (hunk 2 above)
Comment (stu1130) [reply examples/build.gradle:74]: I will collect the benchmark data

**Comment by stu1130** [comment]
close the PR as the benchmark data is pretty bad

**Pull request closed without merging**